pyproject.toml

```diff
@@ -12,21 +12,15 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = "^3.8"
-click = "*"
-importlib-metadata = ">=4.8.0"
-pyparsing = "^3.0.9"
-bioregistry = ">=0.9.43"
-deprecation = ">=2.1.0"
-linkml-runtime = ">=1.5.3"
+linkml-runtime = ">=1.5.5"
+importlib-metadata = ">=6.8.0"
+pansql = ">=0.0.1"
+sssom-schema = ">=0.14.0"
+bioregistry = ">=0.9.87"
 networkx = ">=3.1"
-curies = ">=0.5.7"
```

Comment:
I don't think this makes total sense - `curies` is a transitive dependency of other dependencies of sssom-py, but we are using 0.5.7+ only features, so this needs to be annotated. I think the rule of thumb is you should mark everything you explicitly import as explicit dependencies

Reply:
Same for pandas pyyaml (and perhaps others) - whenever we directly use a library we should add it here.

Reply:
I am starting with the minimum possible dependency list to see where the error is occurring and also satisfies the tests to pass. This is still a work in progress and nothing is final yet. Be rest assured everything will be put back.

Reply:
@cthoyt and @matentzn are correct, if you depend on it, declare it. Just because it is a transitive dependency today does not mean it will be tomorrow.

What you should be doing is refactoring the code so these are not  mandatory direct requirements. There are standard patterns here. E.g move to extras and do a dynamic import.

We need to do this. This library has too many dependencies for a library that should be providing minimal core functionality for a simple standard